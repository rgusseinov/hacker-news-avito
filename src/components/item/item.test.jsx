import React from "react";
import renderer from "react-test-renderer";
import Item from "./item";

describe('item component', () => {
  
  const item = {
    id: 1,
    score: 0,
    url: 'some url here',
    title: 'Here the name of article',
    by: 'Ruslan',
    descendants: 50
  };

  it(`<Item /> should render`, () => {
    const tree = renderer
    .create(<Item
      item={item}
    />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});

