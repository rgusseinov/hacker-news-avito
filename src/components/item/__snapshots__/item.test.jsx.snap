// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`item component <Item /> should render 1`] = `
<div
  className="news-item"
>
  <div
    className="news-item__box-rating"
  >
    <div
      className="rating up"
    >
      0
    </div>
    <span>
      Points
    </span>
  </div>
  <div
    className="news-item__box-content"
  >
    <a
      href="http://abc.com"
      rel="noreferrer"
      target="_blank"
    >
      <h2>
         
        Here the name of article
         
      </h2>
    </a>
    <div
      className="news-item__box-footer"
    >
      <ul>
        <li>
          <i
            className="icon icon-user"
          />
          <span>
             
            Ruslan
             
          </span>
        </li>
        <li>
          <a
            href="/item/1"
            onClick={[Function]}
          >
            <i
              className="icon icon-comment"
            />
            <span>
               
              50
               Comments 
            </span>
          </a>
        </li>
        <li>
          <i
            className="icon icon-time"
          />
          <span>
             
            0 days
             
          </span>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="news-item__box-action"
  >
    <a
      href="http://abc.com"
      rel="noreferrer"
      target="_blank"
    >
      Open link
    </a>
  </div>
</div>
`;
